packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/WorkspaceApp/WorkspaceAppModal.svelte

```diff
@@ -138,35 +145,69 @@
   }
 
   $: if (isNew && data?.name && !validationState.touched.url) {
-    data.url = `/${data.name.toLowerCase().replace(/\s+/g, "-")}`
+    data.url = `/${data.name
+      .toLowerCase()
+      .trim()
+      .replace(/[^a-zA-Z0-9 ]/g, "")
+      .replace(/\s+/g, "-")
+      .replace(/-+/g, "-")}`
   }
+
+  $: editingPublishedApp =
+    workspaceApp?.publishStatus.state === PublishResourceState.PUBLISHED
 </script>
 
 <Modal bind:this={modal} on:show={onShow} on:hide>
-  <ModalContent {title} {onConfirm} size="M">
+  <ModalContent {title} {onConfirm} size="M" disabled={editingPublishedApp}>
```

Comment:
This should probably be done in the server as well, but it's not really critical for now